### PR TITLE
services/horizon: Get /ledger/{ledger_id} fixes and refactorings

### PR DESCRIPTION
**Pull request by 2opremio**
Follow up to #2829 . Part of #2582

### CODE REVIEW

File: services/horizon/internal/actions/helpers.go [hunk 1]
```diff
@@ -724,14 +724,6 @@ func GetParams(dst interface{}, r *http.Request) error {
 			if key == "*" {
 				continue
 			}
-			val := query.Get(key)
-			if len(val) > 0 {
```
Comment (abuiles) [inline services/horizon/internal/actions/helpers.go:728]: @2opremio what was the reason for this again? I think we want to explicitly do this so you don't end up with scenarios like the one you describe in slack `/foo/param1/bar?param1=baz`

Comment (2opremio) [reply services/horizon/internal/actions/helpers.go:728]: Why would we treat the query parameter key matching our (internal) URL path parameter placeholder specially?

The user can already supply other unused query parameters and we don't complain abiut. it.

Comment (2opremio) [reply services/horizon/internal/actions/helpers.go:728]: With this change we still do the right thing (use URL the parameter).

Comment (abuiles) [reply services/horizon/internal/actions/helpers.go:728]: @2opremio can you do a test -- what happens if you  send a request to `/offers/:offer_id` with 2 valid offer ids (both different) in the following form: `/offers/1234?offer_id=12345` -- does it server `1234` or `12345` ? I think that's why we have this check

Comment (2opremio) [reply services/horizon/internal/actions/helpers.go:728]: I already added a test for this. Or rather, I modified an existing one.

**Pull request merged**